### PR TITLE
Replace code duplicates with closeAllSuppress helper

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSource.java [hunk 1]
```diff
@@ -283,16 +284,7 @@ public Metrics getMetrics()
...
-        try {
-            close();
-        }
-        catch (RuntimeException e) {
-            // Self-suppression not permitted
-            if (throwable != e) {
-                throwable.addSuppressed(e);
-            }
-        }
+        closeAllSuppress(throwable, this);
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSource.java:287]: remove the `closeWithSuppression` wrapper

**Review by findepi — APPROVED** [review]
Thanks!

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/Closables.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+public final class Closables
+{
+    private Closables() {}
+
+    public static <T extends Throwable> T closeAllSuppress(T rootCause, AutoCloseable... closeables)
+    {
+        requireNonNull(rootCause, "rootCause is null");
+        if (closeables == null) {
+            return rootCause;
```
[the remark this answers is not in the document]
Comment (losipiuk) [reply lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/Closables.java:26]: good point

[the remark this answers is not in the document]
Comment (losipiuk) [reply lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/Closables.java]: Added some tests too.

File: lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/util/ClosablesTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        closeAllSuppress(rootException, nonThrowingClosable);
+        assertTrue(nonThrowingClosable.isClosed());
+        assertThat(rootException.getSuppressed()).isEmpty();
+    }
+
+    @Test
+    public void testCloseAllSuppressNullClosable()
+    {
+        RuntimeException rootException = new RuntimeException("root");
+        closeAllSuppress(rootException, (AutoCloseable) null);
```
Comment (losipiuk) [inline lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/util/ClosablesTest.java:62]: @findepi If I passed just `null` here (without the cast) it was internally interpreted as whole `closeables` array and `requireNonNull(closeables, "closeables is null")` check trigger. 
I do not think this is a big deal though. Why would someone use untyped null literal here?

Comment (findepi) [reply lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/util/ClosablesTest.java:62]: > I do not think this is a big deal though. Why would someone use untyped null literal here?

agreed, that would be wrong

**Pull request merged**